teps/0077-pipeline-partial-execution.md

```diff
@@ -0,0 +1,330 @@
+- [Proposal](#proposal)
+    - [Notes/Caveats (optional)](#notescaveats-optional)
+    - [Risks and Mitigations](#risks-and-mitigations)
+    - [User Experience (optional)](#user-experience-optional)
+    - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Add an ability for `PipelineRun` to have disabled tasks i.e. a `PipelineRun`
+can execute a `Pipeline` partially. 
+
+Allow `PipelineRun` to be created from previous `PipelineRun`.
+
+So, a `PipelineRun` can be partially run or cancelled at run time, and
+resumed at a later point with the help of work proposed in this TEP.
+
+Together these will bring in the ability to resume/retry a failed `PipelineRun`
+
+## Motivation
+
+1. A very long Pipeline may fail due to a transient failure, and the user may
+   want to only rerun the `Tasks` that failed. This is the most significant
+   motivation, as we use tekton as a backend for running our machine learning
+   pipelines at Kubeflow.
+2. It is not enough to `retry` a `PipelineTask` n times, as the failures can
+   be due to e.g. service outage. A manual resume/ retry may be helpful.
+3. Iterate quickly, by disabling tasks that take longer time. This can be done
+   at run time, i.e. without editing pipeline definition.
+
+### Goals
+
+- Partially execute a `Pipeline`, by disabling tasks.
+- Resume a failed or cancelled `PipelineRun` from previous `PipelineRun`.
+- Discuss how the results/workspaces of disabled tasks can be referenced or
+  populated from previous `PipelineRun`.
+
+### Non-Goals
+
+- TBD.
+
+### Use Cases (optional)
+
+1. *Optimal use of resources*: `tektoncd` as a backend for ML. 
+   A machine learning pipeline may consist of tasks moving large amount of
+   data and then training ml models, all of it can be very resource consuming
+   and inability to retry would require a user to start the entire pipeline
+   over. A manual retry, with the ability to specify what tasks should
+   be skipped, may be helpful.
+2. Partial execution of pipeline. This is useful for reusing an existing
+   pipeline i.e. a user can disable certain task from an existing pipeline
+   and in this way run it without creating a new pipeline.
+3. Partial execution is also helpful for testing, i.e. skipping some tasks
+   and developing and testing iteratively and quickly.
+4. Pause and resume, i.e. one could manually cancel a running `PipelineRun`
+   and resume at later point.
+
+## Requirements
+- Create a new `PipelineRun` to resume or retry a completed `PipelineRun`.
```

Comment:
we also have a requirement for the partial execution of a `pipeline`, where there's no previous `pipelinerun`

(represented in use case 2 above)

Reply:
Use case 2 and the requirement 2 are of less importance for us. But, we are happy(more than happy) to pursue them, incase you think we should drop them for this proposal, then I can move the `disableTasks` field under `pipelineRunRef` field.